mrp_timesheet_link/models/mrp_production.py

```diff
@@ -0,0 +1,56 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields, api, _
+
+
+class MRPProduction(models.Model):
+    _inherit = "mrp.production"
+
+    project_id = fields.Many2one(comodel_name='project.project')
+    task_id = fields.Many2one(
+        comodel_name='project.task',
+        inverse='_set_task_id'
+    )
+
+    timesheet_ids = fields.Many2many('account.analytic.line', compute="_compute_timesheet_ids", store=True)
+    timesheet_count = fields.Integer(compute="_compute_timesheet_count")
+
+    @api.onchange('project_id')
+    def _onchange_project_id(self):
+        self.task_id = False
+
+    @api.depends('task_id', 'task_id.timesheet_ids')
+    def _compute_timesheet_ids(self):
+        for record in self:
+            record.timesheet_ids = record.task_id.timesheet_ids
+            record.timesheet_count = len(record.task_id.timesheet_ids)
+
+    def _compute_timesheet_count(self):
+        for record in self:
+            record.timesheet_count = len(record.timesheet_ids)
```

Comment:
This could achived without need of `timesheet_ids` (production timesheets are actually task timesheets):
```suggestion
            record.timesheet_count = len(record.task_id.timesheet_ids)
```